hudi-common/src/main/java/org/apache/hudi/common/model/HoodieIndexDefinition.java

```diff
@@ -183,8 +226,25 @@ public Builder withIndexOptions(Map<String, String> indexOptions) {
       return this;
     }
 
+    public Builder withVersion(HoodieIndexVersion version) {
+      if (indexName == null) {
+        throw new IllegalStateException("Please set index name first before setting version");
+      }
+      // Make sure the version enum matching the metadata partition is used.
+      version.ensureVersionCanBeAssignedToIndexType(MetadataPartitionType.fromPartitionPath(indexName));
```

Comment:
hmmm. this predates this PR.. but should n't we be able to directly use `indexType` to check version